legal-api/src/legal_api/resources/business/business_filings.py

```diff
@@ -556,7 +556,7 @@ def _create_invoice(business: Business,  # pylint: disable=too-many-locals
 
         else:
             mailing_address = business.mailing_address.one_or_none()
-            corp_type = filing.json['filing']['business'].get('legalType', business.identifier[:-7])
+            corp_type = filing.json['filing']['business'].get('legalType', business.legal_type)
```

Comment:
nice catch -- can we actually make this the other way around though? I don't think we should trust the json from the front end for legal type unless we have to